torch/_decomp/decompositions.py

```diff
@@ -5091,7 +5091,10 @@ def isin(elements, test_elements, *, assume_unique=False, invert=False):
     if not isinstance(elements, torch.Tensor):
         elements = torch.tensor(elements, device=test_elements.device)
     if not isinstance(test_elements, torch.Tensor):
-        test_elements = torch.tensor(test_elements, device=elements.device)
+        if invert:
+            return torch.ne(elements, test_elements)
+        else:
+            return torch.eq(elements, test_elements)
```

Comment:
Judging by the other change, this should be
```suggestion
        cmp = torch.eq(elements, test_elements)
        return ~cmp if invert else cmp
```

Reply:
Thanks for your suggestion. Since `test_elements` is a scalar tensor in this case, the issue fixed by the other change shouldn't occur here as using `ops.ne`. Additionally, when `invert` is True, the previous implementation uses only one `ops.ne`, whereas the new implementation introduces two operations: `ops.eq` and `ops.logical_not`. Therefore, I feel the previous implementation is more efficient. Looking forward to your further comments.

Reply:
I think this gives a different result for floating point nans, `(x == y) != !(x != y)` when either `x` or `y` is NaN.

Reply:
> (x == y) != !(x != y) when either x or y is NaN.

Not sure if my understanding is right. But It seems to return `False` in C++, which appears to treat `(x == y)` as equivalent to `!(x != y)` when either `x` or `y` is `NaN`.

```
double x = std::nan("");
double y = 1.0;
std::cout<< ((x == y)) <<std::endl;
std::cout<< (!(x != y)) <<std::endl;
std::cout<< ((x == y) != !(x != y)) <<std::endl;
std::cout<< ((x == y) == !(x != y)) <<std::endl;
---------------

0
0
0
1
```

Reply:
Ah okay, the expected condition doesn't hold for `<` and `>=`  and I extrapolated too far. So why did you make that change in `isin_default` then? What am I missing.

Reply:
Ah, the previous implementation has a different result as eager when the `elements = 1`, `test_elements = torch.tensor([1, 2, 3, 4])` and `invert`

- `(x != test_elements).any(dim=dim)` returns `True`
- `~((x == test_elements).any(dim=dim))` returns `False`

Reply:
Ah of course, because `~any(x) <=> all(~x)` and not `any(~x)`.